crypto/lhash/lhash.c

```diff
@@ -41,10 +43,12 @@ OPENSSL_LHASH *OPENSSL_LH_new(OPENSSL_LH_HASHFUNC h, OPENSSL_LH_COMPFUNC c)
     ret->down_load = DOWN_LOAD;
     return (ret);
 
+ err2:
```

Comment:
free of a  NULL pointer is fine.  As long as ret is init'd (via zalloc or explicitly), then you can collapse to one goto label.